engine/src/main/java/org/hibernate/search/engine/search/aggregation/dsl/RangeAggregationFieldStep.java

```diff
@@ -35,9 +35,27 @@ default <F> RangeAggregationRangeStep<?, PDF, F> field(String fieldPath, Class<F
 	 * @param type The type of field values.
 	 * @param <F> The type of field values.
 	 * @param convert Controls how the ranges passed to the next steps and fetched from the backend should be converted.
-	 * See {@link ValueConvert}.
+	 * See {@link org.hibernate.search.engine.search.common.ValueConvert}.
 	 * @return The next step.
+	 * @deprecated Use {@link #field(String, Class, ValueModel)} instead.
 	 */
-	<F> RangeAggregationRangeStep<?, PDF, F> field(String fieldPath, Class<F> type, ValueConvert convert);
+	@Deprecated
+	default <F> RangeAggregationRangeStep<?, PDF, F> field(String fieldPath, Class<F> type,
+			org.hibernate.search.engine.search.common.ValueConvert convert) {
+		return field( fieldPath, type,
+				org.hibernate.search.engine.search.common.ValueConvert.toValueModel( convert ) );
+	}
+
+	/**
+	 * Target the given field in the range aggregation.
+	 *
+	 * @param fieldPath The <a href="SearchAggregationFactory.html#field-paths">path</a> to the index field to aggregate.
+	 * @param type The type of field values.
+	 * @param <F> The type of field values.
+	 * @param convert Controls how the ranges passed to the next steps and fetched from the backend should be converted.
+	 * See {@link ValueModel}.
+	 * @return The next step.
+	 */
+	<F> RangeAggregationRangeStep<?, PDF, F> field(String fieldPath, Class<F> type, ValueModel convert);
```

Comment:
```suggestion
	 * @param valueModel The model of range values, used to determine how range values passed to the next steps and fetched from the backend should be converted.
	 * See {@link ValueModel}.
	 * @return The next step.
	 */
	<F> RangeAggregationRangeStep<?, PDF, F> field(String fieldPath, Class<F> type, ValueModel valueModel);
```

This is probably something to fix in other places too.